envisage/service.py

```diff
@@ -74,7 +74,7 @@ def get(self, obj, trait_name):
     def set(self, obj, name, value):
         """ Trait type setter. """
 
-        raise SystemError("Service traits cannot be set")
+        raise RuntimeError("Service traits cannot be set")
```

Comment:
This should be a `TraitError`. (Note: the suggestion below probably isn't enough by itself - you'll also need to import `TraitError`.)
```suggestion
        raise TraitError("Service traits cannot be set")
```